### PR TITLE
Add `username` and `password` options

### LINKED ISSUE #352: Add Credentials parameter in redis_nooproc
### What action do you want to perform
I want to conncect to an existing redis server, which requires authentication

### What are the results
Currently I can only specify `host` and `port` to `factories.redis_nooproc`

### What are the expected results
By supplying additional parameters `username` and `password` to `factories.redis_nooproc`, these parameters are passed further down to the `redis.Redis` instance, which will then automatically authenticate the client

### PR DESCRIPTION
**Pull request by tibotix**
Fixes #352 .

See issue and commit messages for more detail.

### CODE REVIEW

File: src/pytest_redis/executor.py [hunk 1]
```diff
@@ -115,6 +119,9 @@ def __init__(
         self.unixsocket = str(datadir / f"redis.{port}.sock")
         self.executable = executable
 
+        self.username = username
```
Comment (fizyk) [inline src/pytest_redis/executor.py:122]: username isn't used for RedisExecutor

File: src/pytest_redis/factories.py [hunk 2]
```diff
@@ -200,6 +208,8 @@ def redisdb_factory(request):
 
         redis_host = proc_fixture.host
         redis_port = proc_fixture.port
+        redis_username = proc_fixture.username if hasattr(proc_fixture, "username") else None
+        redis_password = proc_fixture.password if hasattr(proc_fixture, "password") else None
```
Comment (fizyk) [inline src/pytest_redis/factories.py:212]: at this moment, all proc_fixtures have password attribute (and username, but regular executor does not actually needs it)

**Review by fizyk — CHANGES_REQUESTED** [review]
Thank you for taking care of readme as well :)

File: src/pytest_redis/executor.py (hunk 1 above)
Comment (tibotix) [reply src/pytest_redis/executor.py:122]: yes i set it for consistency reasons so all executors have a username/password attribute.

File: src/pytest_redis/factories.py (hunk 2 above)
Comment (tibotix) [reply src/pytest_redis/factories.py:212]: I can either leave the username attribute in regular executor and remove the hasattr checks, or remove the username attribute in regular executor and leave the hasattr at least for the username

**Comment by tibotix** [comment]
i removed the `hasattr` checks in `factories.py`. Let me know what you think :)

**Review by fizyk — CHANGES_REQUESTED** [review]
@tibotix sorry for the time it took me to respond.

There seems to be some formatting issues, that black is reporting.

The other thing is I just merged the pr that adds wait for the NoopRedis and changes it from the NamedTuple to actual class.

**Comment by tibotix** [comment]
Alright, I will take a look at it :)

**Pull request merged**